### PR TITLE
fix: split apart big Cmake libraries

### PR DESCRIPTION
**Pull request by Xiphoseer**
This PR introduces a bunch of new static libraries so that `make -j` can schedule more work in parallel. This is mostly achieved by using `target_link_libraries` with `INTERFACE` i.e. requiring modules that depend on the current library to link the sibling modules without waiting for that lib to build (to link against it, which makes little sense for static libs) and doing some of the include_dir handling differently.

### CODE REVIEW

**Comment by EmosewaMC** [comment]
current test results
make clean && time make -j4
new
real    2m5.752s
user    5m11.703s
sys     2m28.125s

old
real    2m18.390s
user    4m57.250s
sys     3m44.563s

save Entity.h (a pch header) && time make -j4
new
real    1m19.901s
user    3m2.547s
sys     1m56.984s

old
real    1m18.705s
user    3m5.406s
sys     1m24.547s

Adding a new script (dScripts/client/ai/PR/NewScript.(cpp/h)) && time make -j4
new
real    0m14.619s
user    0m5.500s
sys     0m5.625s

old
real    1m20.967s
user    2m52.047s
sys     1m45.641s

Save InventoryComponent.h && time make -j4
new
real    0m21.726s
user    0m50.828s
sys     0m17.969s

old
real    0m21.473s
user    0m50.094s
sys     0m23.813s

File: dGame/dEntity/CMakeLists.txt [hunk 1]
```diff
@@ -1,2 +1,6 @@
-set(DGAME_DENTITY_SOURCES "EntityCallbackTimer.cpp"
-	"EntityTimer.cpp" PARENT_SCOPE)
+set(DGAME_DENTITY_SOURCES
+	"EntityCallbackTimer.cpp"
+	"EntityTimer.cpp")
+
+add_library(dEntity STATIC ${DGAME_DENTITY_SOURCES})
+target_precompile_headers(dEntity REUSE_FROM dGameBase)
```
Comment (Xiphoseer) [inline dGame/dEntity/CMakeLists.txt:6]: Missing target_include_directories

File: dGame/dBehaviors/CMakeLists.txt [hunk 2]
```diff
@@ -54,4 +54,9 @@ set(DGAME_DBEHAVIORS_SOURCES "AirMovementBehavior.cpp"
 	"TargetCasterBehavior.cpp"
 	"TauntBehavior.cpp"
 	"VentureVisionBehavior.cpp"
-	"VerifyBehavior.cpp" PARENT_SCOPE)
+	"VerifyBehavior.cpp")
+
+add_library(dBehaviors STATIC ${DGAME_DBEHAVIORS_SOURCES})
+target_link_libraries(dBehaviors
```
Comment (Xiphoseer) [inline dGame/dBehaviors/CMakeLists.txt:60]: Missing target_include_directories

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]
one nit

File: dGame/dComponents/CMakeLists.txt [hunk 3]
```diff
@@ -46,5 +47,11 @@ set(DGAME_DCOMPONENTS_SOURCES "ActivityComponent.cpp"
...
 	"MiniGameControlComponent.cpp"
-	PARENT_SCOPE
 )
+
+add_library(dComponents STATIC ${DGAME_DCOMPONENTS_SOURCES})
+target_include_directories(dComponents PRIVATE ${PROJECT_SOURCE_DIR}/dScripts/02_server/Map/General) # PetDigServer.h
+target_precompile_headers(dComponents REUSE_FROM dGameBase)
+target_link_libraries(dComponents
+	PUBLIC dPhysics dDatabase
+	INTERFACE dUtilities dCommon dBehaviors dChatFilter dMission dInventory)
```
Comment (EmosewaMC) [inline dGame/dComponents/CMakeLists.txt:57]: nit: newline

**Review by aronwk-aaron — APPROVED** [review]

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**